### PR TITLE
chore: Random ids rework part4

### PR DESCRIPTION
**Pull request by sfc-gh-asawicki**
Do not use the following methods for id creation in tests:
- NewDatabaseObjectIdentifier
- NewDatabaseObjectIdentifierFromFullyQualifiedName
- NewSchemaObjectIdentifierInSchema
- NewSchemaObjectIdentifier
- NewSchemaObjectIdentifierWithArguments
- NewSchemaObjectIdentifierFromFullyQualifiedName

### CODE REVIEW

**Review by sfc-gh-jmichalak — CHANGES_REQUESTED** [review]

File: pkg/resources/grant_database_role_acceptance_test.go [hunk 1]
```diff
@@ -128,9 +127,11 @@ func TestAcc_GrantDatabaseRole_accountRole(t *testing.T) {
...
 // proves https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2410 is fixed
 func TestAcc_GrantDatabaseRole_share(t *testing.T) {
-	databaseName := acc.TestClient().Ids.Alpha()
-	databaseRoleName := acc.TestClient().Ids.Alpha()
-	databaseRoleId := sdk.NewDatabaseObjectIdentifier(databaseName, databaseRoleName).FullyQualifiedName()
+	databaseId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	databaseName := databaseId.Name()
+	databaseRoleId := acc.TestClient().Ids.RandomDatabaseObjectIdentifierInDatabase(databaseId)
+	databaseRoleName := databaseRoleId.Name()
+	databaseRoleFullyQualifiedName := databaseRoleId.FullyQualifiedName()
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/resources/grant_database_role_acceptance_test.go:134]: Inlined in this file

File: pkg/sdk/test_ids_test.go [hunk 2]
```diff
@@ -1,4 +1 @@
 package sdk
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/sdk/test_ids_test.go:1]: done

File: pkg/sdk/testint/external_functions_integration_test.go [hunk 3]
```diff
@@ -260,10 +251,9 @@ func TestInt_ExternalFunctions(t *testing.T) {
 	})
 
 	t.Run("describe external function", func(t *testing.T) {
-		e := createExternalFunction(t)
-		id := e.ID()
+		e, _ := createExternalFunction(t)
 
-		request := sdk.NewDescribeExternalFunctionRequest(id, []sdk.DataType{sdk.DataTypeVARCHAR})
+		request := sdk.NewDescribeExternalFunctionRequest(e.ID(), []sdk.DataType{sdk.DataTypeVARCHAR})
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/sdk/testint/external_functions_integration_test.go:256]: Check SNOW-999049 - `consider NewSchemaObjectIdentifierWithArguments problems (e.g. <object>.ID() is not returning the arguments so ids are not equal)`

**Review by sfc-gh-jcieslak — DISMISSED** [review]

File: pkg/sdk/random_test.go [hunk 4]
```diff
@@ -6,7 +6,12 @@ import (
 
 var (
 	invalidAccountObjectIdentifier = NewAccountObjectIdentifier(random.StringN(256))
-	invalidSchemaObjectIdentifier  = NewSchemaObjectIdentifier(random.StringN(255), random.StringN(255), random.StringN(255))
+	longSchemaObjectIdentifier     = NewSchemaObjectIdentifier(random.StringN(255), random.StringN(255), random.StringN(255))
+
+	// TODO: Add to the generator
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/sdk/random_test.go:11]: Yup, Added.

**Review by sfc-gh-jmichalak — APPROVED** [review]

**Review by sfc-gh-jcieslak — DISMISSED** [review]

**Review by sfc-gh-jcieslak — APPROVED** [review]

**Pull request merged**